### ISSUE TITLE
IdsDropdown: Cannot have empty string as value

### ISSUE THREAD

**Issue by anhallbe** [infor-design/enterprise-wc#1838]
**Describe the bug**
If an `<ids-dropdown>` has an "option" (`<ids-list-box-item>`) with a value of `""`, that item is not selectable.

I.e this code:

```html
      <ids-dropdown value="" label="Drodown with empty value">
        <ids-list-box>
          <ids-list-box-option value="">Empty</ids-list-box-option>
          <ids-list-box-option value="alice">Alice</ids-list-box-option>
          <ids-list-box-option value="bob">Bob</ids-list-box-option>
        </ids-list-box>
      </ids-dropdown>
```

**To Reproduce**

1. Go to https://stackblitz.com/edit/web-platform-vnw841?file=index.html
2. Click on the dropdown and select a value, e.g Alice
3. Click on the dropdown and select the initial value, i.e Empty
4. The dropdown still has Alice as the selected value

**Expected behavior**
The Empty value should be selected. This is how the `<select>` element works, and is how the `soho-dropdown` works.

**Version**

- ids-enterprise-wc: 1.0.0-beta.19

**Screenshots**

https://github.com/infor-design/enterprise-wc/assets/4991954/08808d20-7082-4839-9afc-4f33d8ccc8a5

**Platform**
 - Infor Application/Team Name: Portal,  Homepages

**Additional context**

**Comment by tmcconechy** [comment]
I made one example with a blank item: https://main.wc.design.infor.com/ids-dropdown/example.html
With a distinct setting https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-dropdown/demos/example.html#L141

But if this is also needed to have a blank value we can also make that work.

**Comment by anhallbe** [comment]
Yeah the blank value might seem like an edge case, but we use it all the time to represent a "no preference" / "default" in our server-side models.

For instance, the model for "selected language" could be one of `"en-US"`, `"sv-SE"`, or `""` (representing the browser default).

To make this work with the current dropdown, we'd have to transform the empty value to something for the Dropdown control (e.g `"" -> "browser-default"`, and then back again when sending it off to the server (`"browser-default" -> ""`). It's not a huge issue, but it is a bit annoying especially since it deviates from how `<select>`/`soho-dropdown` works. So allowing empty strings as value would be much appreciated :)

**Comment by jbrcna** [comment]
QA Passed
<img width="1653" alt="image" src="https://github.com/infor-design/enterprise-wc/assets/56722184/0e348d56-07a6-40c6-8c5c-0c9db4682d5d">
